### DISCUSSION TITLE
Handling in AbstractUIExtensions

### DISCUSSION THREAD

**Discussion by JValGLZ** [eclipse-glsp/glsp#1301] [Q&A]
Hello, I am working on the Java-EMF-Theia Example and I have made a UIExtension. The UI also implements the IActionHandler. I wanted to set up a simple handle function which is the following. 

```typescript
handle(action: Action): void | Action | ICommand {
        console.log('Handling action: ' + action);
    }
``` 
I have already previously defined other actions and when they occured, I figured that this handle function would print out the statement, but it does not. What is it that I am missing? Thank you for your help.

**Comment by tortmayr** [comment]
@JValGLZ 
Simply implementing the `IActionHandler` interface is not enough. In addition, it needs to be configured in the DI module with
the `configureActionHandler` method. Here you define which action kinds the `IActionHandler` should handle.
E.g:

```ts
const diagramModule = new ContainerModule((bind, _unbind, isBound, rebind) => {
    const context = { bind, _unbind, isBound, rebind };
    bind(MyUIExtension).toSelf().inSingletonScope();
    bind(TYPES.IUIExtension).toService(MyUIExtension);
    configureActionHandler(context, SomeAction.KIND, MyUIExtension);
    configureActionHandler(context, SomeOtherAction.KIND, MyUIExtension);
}
```
(See also https://eclipse.dev/glsp/documentation/actionhandler/#implementing-an-action-handler-glsp-client)

**Reply by JValGLZ** [reply]
Hello! Apologies for the late reply. What I really mean is that I have created a dialog and have set up my own CreateNode operations as well as creating transitions with these nodes. Since these are two separate operations, I wanted to know what would be the way so that this dialog can change information based on both of the operations. I thought up the idea of making both operations return the same action so I can handle it from the client dialog I made, but I am not sure if that is possible.  

Initially, I thought that it would be possible to make it so that the dialog handles all actions from the panel and then check for what type of action it is so that it reacts to only a certain one. I checked the return type of operations which is a command and was unsure how to go about with my idea. 

Please let me know the correct process to do this.

Thank you!
